File: src/main/java/me/jameschan/burrow/command/builtin/RootCommand.java
package me.jameschan.burrow.command.builtin;

import me.jameschan.burrow.command.Command;
import me.jameschan.burrow.context.RequestContext;
import picocli.CommandLine;

@CommandLine.Command(
    name = "root",
    mixinStandardHelpOptions = true,
    description = "Prints the root directory of the application."
)
public class RootCommand extends Command {
    public RootCommand(final RequestContext context) {
        super(context);
    }

    @Override
    public Integer call() {
        buffer.append(context.getRootDir().toString());

        return 0;
    }
}

File: src/main/kotlin/carton/standard/RootCommand.kt
package burrow.carton.standard

import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine

@CommandLine.Command(
    name = "root",
    description = [
        "Displays the absolute path to the root directory of the current " +
                "chamber."
    ]
)
class RootCommand(data: CommandData) : Command(data) {
    override fun call(): Int {
        stdout.println(chamber.rootPath)

        return CommandLine.ExitCode.OK
    }
}